firmware/applications/SignBuddy/imu.c

```diff
@@ -48,6 +49,20 @@ static void hw_init(void)
   i2c_init(&s.i2c_instance, IMU_I2C, &i2c_config);
 }
 
+static inline void sample_data(void)
+{
+  ulTaskNotifyTake(pdFALSE, portMAX_DELAY);
+  uint8_t ret = 0;
+
+  ret |= bno055_get_sys_calib_stat(&s.sys_calib_stat);
+  ret |= bno055_get_mag_calib_stat(&s.mag_calib_stat);
+  ret |= bno055_get_accel_calib_stat(&s.accel_calib_stat);
+  ret |= bno055_get_gyro_calib_stat(&s.gyro_calib_stat);
```

Comment:
I don't think we should include the status values in the sample data

Reply:
yeah, agreed

Reply:
okay, i will remove these. Do you know of any way to iterate over a struct so i don't have to create a separate array to store values?

Reply:
Store the values directly in a struct of this type 
![image](https://user-images.githubusercontent.com/55762060/153065192-f3e7fd81-1862-45ef-aea2-27f6a4d56101.png)

Reply:
You can iterate over a struct with a pointer

Reply:
you can look at this code now to see how I would do it